src/main/java/org/ibp/api/java/impl/middleware/study/StudyEntryServiceImpl.java

```diff
@@ -91,13 +117,60 @@ public StudyEntryDto replaceStudyEntry(final Integer studyId, final Integer entr
 			.replaceStudyEntry(studyId, entryId, gid, this.pedigreeService.getCrossExpansion(gid, this.crossExpansionProperties));
 	}
 
+	@Override
+	public List<StudyEntryDto> createStudyEntries(final Integer studyId,
+		final StudyEntryGeneratorBatchRequestDto studyEntryGeneratorBatchRequestDto, final String programUuid) {
+		this.studyValidator.validate(studyId, true);
+		if(studyEntryGeneratorBatchRequestDto.getListId() != null && studyEntryGeneratorBatchRequestDto.getListId() != 0) {
+			return this.createStudyEntries(studyId, studyEntryGeneratorBatchRequestDto.getListId());
+		}
+
+		//Validate EntryType
+		this.entryTypeValidator.validateEntryType(studyEntryGeneratorBatchRequestDto.getEntryTypeId(), programUuid);
```

Comment:
Instead of declaring programUUID as method parameter and passing to `EntryTypeValidator`, you can use `ContextHolder.getCurrentProgram()` from `EntryTypeValidator` class